Mathlib/NumberTheory/ModularForms/Identities.lean

```diff
@@ -0,0 +1,31 @@
+/-
+Copyright (c) 2024 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+
+import Mathlib.NumberTheory.ModularForms.SlashInvariantForms
+import Mathlib.NumberTheory.ModularForms.CongruenceSubgroups
+
+/-!
+# Identities of ModularForms and SlashInvariantForms
+Collection of useful identities of modular forms.
+-/
+
+noncomputable section
+
+open ModularForm UpperHalfPlane Matrix
+
+namespace SlashInvariantForm
+
+theorem lvl_N_periodic (N : ℕ) (k : ℤ) (f : SlashInvariantForm (Gamma N) k) (z : ℍ) (n : ℤ):
```

Comment:
```suggestion
theorem lvl_N_periodic (N : ℕ) (k : ℤ) (f : SlashInvariantForm (Gamma N) k) (z : ℍ) (n : ℤ) :
```

Reply:
I don't really like the name `lvl_N_periodic`, since it refers to the variable `N`. (Just like I'm not a fan of the name `padic_int`.)

This name suggest that you maybe want a definition of what it means to be `LevelPeriodic`? And then you can prove with this lemma that `SlashInvariantForm`s satisfy that predicate?

Reply:
Hmm good point, so `LevelPeriodic` should be the `width` period, but we dont yet have the widths of cusps ( because we dont have cusps yet). How about I put width in the name (instead of N) for now and once we have cusps I can update the names if needed as this will be a special case of a more general result for any level?